server/lib/realtime/subscribers_notification.ex

```diff
@@ -41,107 +36,112 @@ defmodule Realtime.SubscribersNotification do
+
+              is_map(record) &&
+                Enum.each(record, fn {k, v} ->
+                  should_notify_column = has_column(event_config, schema, table, k)
+
+                  if is_valid_notification_key(v) and should_notify_column do
+                    [table_topic, ":", k, "=eq.", v]
+                    |> IO.iodata_to_binary()
+                    |> RealtimeChannel.handle_realtime_transaction(change)
+                  end
+                end)
+
+            "DELETE" ->
+              old_record = Map.get(change, :old_record)
+
+              is_map(old_record) &&
+                Enum.each(old_record, fn {k, v} ->
+                  should_notify_column = has_column(event_config, schema, table, k)
+
+                  if is_valid_notification_key(v) and should_notify_column do
+                    [table_topic, ":", k, "=eq.", v]
+                    |> IO.iodata_to_binary()
+                    |> RealtimeChannel.handle_realtime_transaction(change)
+                  end
+                end)
+
+            "TRUNCATE" ->
+              nil
           end
 
-        "TRUNCATE" ->
+        _ ->
           nil
       end
-    end
+    end)
   end
 
-  @doc """
-  Determines whether the Realtime config has a specific schema relation
-  """
+  defp notify_subscribers(_txn, _config), do: :ok
+
   defp has_schema(config, schema) do
+    # Determines whether the Realtime config has a specific schema relation
     valid_patterns = ["*", schema]
     Enum.any?(config, fn c -> c.relation in valid_patterns end)
   end
 
-  @doc """
-  Determines whether the Realtime config has a specific table relation
-  """
   defp has_table(config, schema, table) do
+    # Determines whether the Realtime config has a specific table relation
     # Construct an array of valid patterns: "*:*", "public:todos", etc
     valid_patterns =
       for schema_keys <- ["*", schema],
           table_keys <- ["*", table],
-          do: "#{schema_keys}:#{table_keys}"
+          do: [schema_keys, ":", table_keys] |> IO.iodata_to_binary()
 
     Enum.any?(config, fn c -> c.relation in valid_patterns end)
   end
 
-  @doc """
-  Determines whether the Realtime config has a specific column relation
-  """
   defp has_column(config, schema, table, column) do
+    # Determines whether the Realtime config has a specific column relation
     # Construct an array of valid patterns: "*:*:*", "public:todos", etc
     valid_patterns =
       for schema_keys <- ["*", schema],
           table_keys <- ["*", table],
           column_keys <- ["*", column],
-          do: "#{schema_keys}:#{table_keys}:#{column_keys}"
+          do: [schema_keys, ":", table_keys, ":", column_keys] |> IO.iodata_to_binary()
 
     Enum.any?(config, fn c -> c.relation in valid_patterns end)
   end
 
-  defp is_valid_notification_key(v) do
-    v != nil and v != :unchanged_toast and String.length(v) < 100
+  defp is_valid_notification_key(v) when is_binary(v) do
+    String.length(v) < 100
```

Comment:
I'm guessing the `is_binary` covers any cases where Postgres sends out an `:unchanged_toast` event?

Reply:
@kiwicopple yep! I also made sure to have that covered in tests: https://github.com/supabase/realtime/blob/refactor/remove-genservers/server/test/realtime/subscribers_notification_test.exs#L530